koku/api/report/hcs/view.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright 2023 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""View for HCS report posting."""
+# from api.report.hcs.query_handler import HCSReportQueryHandler
+import logging
+from uuid import UUID
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from api.common.permissions.hcs_access import HCSAccessPermission
+from api.report.hcs.serializers import HCSReportSerializer
+from reporting.hcs.models import HCSReport
+
+LOG = logging.getLogger(__name__)
+
+
+class HCSDetailView(APIView):
+    """
+    View to fetch HCS report details for specific provider
```

Comment:
I would use `source` instead of `provider` for external language consistency for the user.

Reply:
I hate the Source/Provider stuff! We say source for a customer but we actually want the Provider UUID for our flow, so confusing! But I agree we can change this :grin: